android/src/main/java/com/reactnative/ivpusic/imagepicker/ImageCropPicker.java

```diff
@@ -356,23 +356,20 @@ private void initiateCamera(Activity activity) {
 
     private void initiatePicker(final Activity activity) {
         try {
-            PickVisualMediaRequest.Builder builder = new PickVisualMediaRequest.Builder();
-            // Simplified media type handling
+            Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+
             if (mediaType.equals("video")) {
-                builder.setMediaType(ActivityResultContracts.PickVisualMedia.VideoOnly.INSTANCE);
+                intent.setType("video/*");
             } else if (mediaType.equals("photo") || cropping) {
                 // Force image-only for cropping
-                builder.setMediaType(ActivityResultContracts.PickVisualMedia.ImageOnly.INSTANCE);
+                intent.setType("image/*");
             } else {
-                builder.setMediaType(ActivityResultContracts.PickVisualMedia.ImageAndVideo.INSTANCE);
+                intent.setType("image/*");
+                intent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*", "video/*"});
             }
 
-            Intent intent;
-            if (multiple) {
-                intent = new ActivityResultContracts.PickMultipleVisualMedia(maxFiles).createIntent(activity, builder.build());
```

Comment:
Note that we're losing the respect of `maxFiles` here. I don't believe that's possible with this approach.